### DISCUSSION TITLE
Half cell flux  calculation over particle/micro scale

### DISCUSSION THREAD

**Discussion by MamunAlSiraj** [pybamm-team/PyBaMM#4491] [Q&A] [answered]
Hello group,

I am newbie in Pybamm and was quite impressed with the capabilites
of Pybamm.

My question is regarding, given pybamm tutorial example with a half cell model
where [-Le<x<0] is electrolyte and [0>x>Lp] is considered positive electrode.

The link for the example is following:
https://docs.pybamm.org/en/latest/source/examples/notebooks/creating_models/5-half-cell-model.html

For my sanity check (to see if I understand the process), I extracted the simulation output values 
such as particle surface concentration (c_s) and the applied flux on
the particle (j_p). I calculate the phi and extracted Phi_p
from the simulation also.
The formulation for surface flux is given as :
jp= 2*jo(c)*sinh(F/(2*R*T)(phi-phi_p-U))

If I extract the value of jp, I see there is minima or maxima.
Under galvanostatic operation, should not the value of
jp be constant ?

Thanks

**Comment by isaacbasil** [comment, marked as answer]
Hi, 
I am not involved in the Pybamm team so someone may come along to correct me... but just from having a look at the example you attached, jp will be a function of both x and time.  Likewise, the inputs to the function jp - namely c_surf, phi_e, and phi_p - are all functions of x and time. Perhaps the intuition that you have of jp being a constant is coming from single-particle models? This model here is not part of this family, and has a new particle at each position on the macroscale (x) domain. 
Plenty of info in this review: 10.1088/2516-1083/ac7d31

**Reply by MamunAlSiraj** [reply]
Hi, thanks for the response and the literature.
I somehow understood that.
